### ISSUE TITLE
Plugin silently fails when mypy `non_interactive` option is set

### ISSUE THREAD

**Issue by DrLex0** [microsoft/vscode-mypy#364]
## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): 3.13.2
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): venv, but probably any
- Operating system (and version): Fedora Linux 41, but again probably any
- Version of tool extension you are using: 2025.2.0

### Expected Behavior
The plugin shows typing errors in the editor

![Image](https://github.com/user-attachments/assets/c7d6c65a-4bf6-42d1-a075-b841cc2def98)

### Actual Behavior
Even the most obvious typing errors do not show up in the editor

![Image](https://github.com/user-attachments/assets/45443af1-fe81-4fbd-87fb-de1f5b58a10e)

## Reproduction Steps:
Add a `setup.cfg` file to the project with this content:
```
[mypy]
install_types = True
non_interactive = True
```

Create a Python file with typing errors:
```Python
def broken() -> bool:
    an_int: int = 42
    a_string = "string"
    kaputt = an_int + a_string
    return kaputt
```
Behold the lack of errors shown in the editor.

## Logs:
First, without the `non_interactive` option, when everything works fine:
```
2025-04-22 17:14:01.400 [info] /bin/python -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --show-error-end /home/someone/DummyProject/mypy_kaputt.py
2025-04-22 17:14:01.401 [info] CWD Server: /home/someone/DummyProject
2025-04-22 17:14:01.677 [info] file:///home/someone/DummyProject/mypy_kaputt.py :
/home/someone/DummyProject/mypy_kaputt.py:4:23:4:30: error: Unsupported operand types for + ("int" and "str")  [operator]
/home/someone/DummyProject/mypy_kaputt.py:5:12:5:17: error: Incompatible return value type (got "int", expected "bool")  [return-value]

2025-04-22 17:14:01.677 [info] file:///home/someone/DummyProject/mypy_kaputt.py :
/home/someone/DummyProject/mypy_kaputt.py:4:23:4:30: error: Unsupported operand types for + ("int" and "str")  [operator]
/home/someone/DummyProject/mypy_kaputt.py:5:12:5:17: error: Incompatible return value type (got "int", expected "bool")  [return-value]
```

Now, with the `non_interactive` option set:
```
2025-04-22 17:11:54.005 [info] /bin/python -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --show-error-end /home/someone/DummyProject/mypy_kaputt.py
2025-04-22 17:11:54.005 [info] CWD Server: /home/someone/DummyProject
2025-04-22 17:11:54.290 [info] /home/someone/DummyProject/mypy_kaputt.py:4:23:4:30: error: Unsupported operand types for + ("int" and "str")  [operator]
/home/someone/DummyProject/mypy_kaputt.py:5:12:5:17: error: Incompatible return value type (got "int", expected "bool")  [return-value]

2025-04-22 17:11:54.291 [info] file:///home/someone/DummyProject/mypy_kaputt.py :
```

Note that even though the typing errors are still in the log, the preceding `file:///` line is absent. It seems the plugin relies on this, even though the file path is also available from the error lines themselves.  
If presence of the `non_interactive` option causing the plugin to not report anything is considered expected behavior, then at least some warning should be shown about this. But ideally, the plugin should just work regardless of this setting.

**Comment by karthiknadig** [comment]
This looks like the output is not getting captured when we run this. The code for this is here:
https://github.com/microsoft/vscode-mypy/blob/8cb6bcadc6798e4449eab864b3f0481f677fcb1a/bundled/tool/lsp_utils.py#L198-L212

**Comment by karthiknadig** [comment]
Looks like in `non_interactive` mode the results are put in `stderr`. So, you see it in logs, but it is not processed. This should be an easy fix. 

Feel free to submit a PR on this if you want the fix sooner. We are open to accepting PRs on this.
